nav2_behavior_tree/plugins/decorator/distance_controller.cpp

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) 2018 Intel Corporation
+// Copyright (c) 2020 Sarthak Mittal
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_BEHAVIOR_TREE__DISTANCE_CONTROLLER_HPP_
+#define NAV2_BEHAVIOR_TREE__DISTANCE_CONTROLLER_HPP_
+
+#include <chrono>
+#include <string>
+#include <memory>
+#include <cmath>
+
+#include "nav2_util/robot_utils.hpp"
+#include "geometry_msgs/msg/pose_stamped.hpp"
+#include "tf2_ros/buffer.h"
+
+#include "behaviortree_cpp_v3/decorator_node.h"
+
+namespace nav2_behavior_tree
+{
+
+class DistanceController : public BT::DecoratorNode
+{
+public:
+  DistanceController(
+    const std::string & name,
+    const BT::NodeConfiguration & conf)
+  : BT::DecoratorNode(name, conf),
+    initialized_(false),
+    distance_(1.0)
+  {
+    getInput("distance", distance_);
+  }
+
+  // Any BT node that accepts parameters must provide a requiredNodeParameters method
+  static BT::PortsList providedPorts()
+  {
+    return {
+      BT::InputPort<double>("distance", 1.0, "Distance")
+    };
+  }
+
+  void initialize()
+  {
+    node_ = config().blackboard->get<rclcpp::Node::SharedPtr>("node");
```

Comment:
Any reason not to put this in the constructor and then not have to have the initialization process at all (or store initialize)?